hudi-table-management-service/src/main/java/org/apache/hudi/table/management/executor/submitter/SparkEngine.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.management.executor.submitter;
+
+import org.apache.hudi.cli.commands.SparkMain;
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.table.management.common.ServiceConfig;
+import org.apache.hudi.table.management.entity.Instance;
+import org.apache.hudi.table.management.entity.InstanceStatus;
+import org.apache.hudi.table.management.exception.HoodieTableManagementException;
+import org.apache.hudi.table.management.store.impl.InstanceService;
+
+import org.apache.spark.launcher.SparkAppHandle;
+import org.apache.spark.launcher.SparkLauncher;
+import org.apache.spark.util.Utils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.spark.launcher.SparkAppHandle.State.FINISHED;
+import static org.apache.spark.launcher.SparkAppHandle.State.SUBMITTED;
+
+public class SparkEngine extends ExecutionEngine {
+
+  private static final Logger LOG = LogManager.getLogger(SparkEngine.class);
+
+  public SparkEngine(InstanceService instanceDao) {
+    super(instanceDao);
+  }
+
+  @Override
+  public Map<String, String> getJobParams(Instance instance) {
+    Map<String, String> sparkParams = new HashMap<>();
```

Comment:
We could possibly register a execution engine and config with a seperate API endpoint and use that engineConfig as an input to run a table service. ServiceConfig is not generalized to any engine today - would rather get this as a input through the rest API.